document persistent major version feature (#840)
owncloud/docs-server: modules/admin_manual/pages/configuration/files/file_versioning.adoc

'versions_retention_obligation' => '30, 30',
----

== Restoring of a Version

> In the WebUI in the details view, there is a restore button next to the versions. It can be used to restore file to its past version. Restore does not delete old version, instead newer version is created.

Reviewer: We might need to add that the retention policy is applied after restoring the file, and the retention policy might remove the old file. If not commented, the user could see the restored file deleted on some cases, and he could think it's a bug.

After adding the author, the version screen changes in the following way:

image::configuration/files/files_versions/concious-persistent-major-versions.png[Author of a changed saved file,width=250]

=== Conscious Persistent Major Versions

ownCloud can automatically generate version numbers for file versions. You can enable and disable to show the author of a version in the versions history via an entry in config.php. For details see: xref:configuration/server/config_sample_php_parameters.adoc#save-extended-version-metadata-of-uploaded-and-edited-files[Save extended version metadata of uploaded and edited files]. Based on this feature, one can define the actual version to be a major version which gets a major version number and all new upcoming versions start counting from this major number. To do so, the user has to click on the check mark on the left of the actual version to publish it as a major version. Major versions are persistent versions that are not subject to any retention policies set by ownCloud. Publishing the version persists it until the original file is deleted. 

> TIP: The versions follow the X.Y format, where X is a major part and Y is a minor part of the version. Version tags cannot be modified by the user and and auto-generated by ownCloud. 

Reviewer: typo: `by the user and and ....`

'versions_retention_obligation' => '30, 30',
----

> == Restoring of a Version

Reviewer 2: ```suggestion
== Restoring a Version
```

> In the WebUI in the details view, there is a restore button next to the versions. It can be used to restore file to its past version. Restore does not delete old version, instead newer version is created. 

Reviewer 3: ```suggestion
In the WebUI in the details view, there is a restore button next to the versions. It can be used to restore the file to a past version. Restore does not delete the old version, instead a new current version is created. 
```

Does "a new current version" make sense? It does to me, but maybe other people have better words for it?

> WARNING: Note that retention policy can be applied after restore operation, which could result in the old version be removed via retention policy. This behaviour can be changed with persistent major version discussed in the section below.

Reviewer 3: ```suggestion
WARNING: Note that retention policy can be applied after the restore operation, which could result in the old version being removed because of the retention policy. This behaviour can be changed with the persistent major version feature discussed in the section below.
```

> WARNING: Note that retention policy can be applied after the restore operation, which could result in the old version being removed because of the retention policy. This behaviour can be changed with the persistent major version feature discussed in the section below.

Reviewer 2: ```suggestion
WARNING: Note that retention policy can be applied after the restore operation, which could result in the old version being removed because of the retention policy. This behavior can be changed with the persistent major version feature discussed in the section below.
```

== Extended Version Metadata

>  You can enable and disable below features via an entry in config.php. For details see: xref:configuration/server/config_sample_php_parameters.adoc#FIXME[Save extended version metadata of uploaded and edited files]. 

Reviewer 3: ```suggestion
 You can enable and disable these features via an entry in config.php. For details see: xref:configuration/server/config_sample_php_parameters.adoc#FIXME[Save extended version metadata of uploaded and edited files]. 
```

> You can enable the feature via an entry in config.php. For details see: xref:configuration/server/config_sample_php_parameters.adoc#save-additional-metadata-for-versions[Save additional metadata for versions]. 

Reviewer 2: ```suggestion
You can enable the feature via an entry in config.php. For details see: xref:configuration/server/config_sample_php_parameters.adoc#save-additional-metadata-for-versions[Save Additional Metadata for Versions]. 
```

image::configuration/files/files_versions/version-metadata.png[Extended Version Metadata,width=300]

=== Show the Author of a Version

> You can enable and disable to show the author of a version in the versions history. This is helpful to keep track of who changed a version. 

Reviewer 3: ```suggestion
You can enable and disable showing the author of a version in the versions history. This is helpful to keep track of who changed a version. 
```

=== Conscious Persistent Major Versions

> ownCloud can automatically generate version numbers for file versions. Based on this feature, one can define the actual version to be a major version which gets a major version number and all new upcoming versions start counting from this major number. To do so, the user has to click on the check mark on the left of the actual version to publish it as a major version. Major versions are persistent versions that are not subject to any retention policies set by ownCloud. Publishing the version persists it until the original file is deleted. 

Reviewer 3: Is the  check mark really on the left?

Reviewer 4: I have created a build and pushed it to staging for "optical" review.

Author: `the user has to click on the check mark on the left of the actual version to publish it as a major version` -> `the user has to click on the check mark next to the version info to publish it as a major version`

the checkmark is on right side...

Reviewer 4: fixed by using the suggested text

> ownCloud can automatically generate version numbers for file versions. Based on this feature, one can define the actual version to be a major version which gets a major version number and all new upcoming versions start counting from this major number. To do so, the user has to click on the check mark on the left of the actual version to publish it as a major version. Major versions are persistent versions that are not subject to any retention policies set by ownCloud. Publishing the version persists it until the original file is deleted. 

Reviewer 4: ```suggestion
ownCloud can automatically generate version numbers for file versions. Based on this feature, one can define the actual version to be a major version which gets a major version number and all new upcoming versions start counting from this major number. To do so, the user has to click on the check mark next to the version info to publish it as a major version. Major versions are persistent versions that are not subject to any retention policies set by ownCloud. Publishing the version persists it until the original file is deleted. 
```

> TIP: The versions follow the X.Y format, where X is a major part and Y is a minor part of the version. Version tags cannot be modified by the user and auto-generated by ownCloud. 

Reviewer 3: ```suggestion
TIP: The versions follow the X.Y format, where X is a major part and Y is a minor part of the version. Version tags cannot be modified by the user, they are auto-generated by ownCloud. 
```
